### PR TITLE
add

### LINKED ISSUE #15981: Add

### PR DESCRIPTION
**Pull request by rohinish404**
closes #15981

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py:258]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py
Comment (ZiadAmerr) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py:260]: @MahmoudAshraf97 Should we use a shared dtype here? Should tensor math functions handle different dtypes?

**Review by ZiadAmerr — CHANGES_REQUESTED** [review]
Amazing work! Just a few changes and if the tests pass we will proceed with the merge :)

**Comment by rohinish404** [comment]
> Amazing work! Just a few changes and if the tests pass we will proceed with the merge :)

Yeah sure!! Do i need to make any changes from my side?

**Comment by ZiadAmerr** [comment]
Nope, LGTM! We will just wait for Mahmoud's approval on this, and if we changing it to `shared_dtypes=False` and the tests pass, we will proceed with the merge :)

**Review by ZiadAmerr — CHANGES_REQUESTED** [review]
Amazing work, @rohinish404! Just this final comment and we are ready to merge.

Sorry for the late review though; I thought I had requested changes this whole time but I have just realized I didn't.

**Comment by rohinish404** [comment]
Should i first have to resolve conflicts?

**Comment by ZiadAmerr** [comment]
No, we can leave that until everything is ready to merge and the checks are passing, then you can resolve the conflicts

**Comment by rohinish404** [comment]
How can i modify the code then? Can't push the changes;)

**Review by ZiadAmerr — APPROVED** [review]
LGTM! Amazing work, thanks a lot @rohinish404, will merge right after the test checks show that the tests are passing :)

**Comment by ZiadAmerr** [comment]
Please resolve the conflicts now :)

**Comment by ZiadAmerr** [comment]
Hey, looks like your function was taken over by another PR due to the comment on the tasks page wasn't in the standard format, you commented `Add` instead of `add` so it was not linked to the actual `add` function in the ToDo tasks. I will merge this PR and then remove the accidental problems myself, thanks a lot for your contribution :)

File: ivy/functional/frontends/paddle/tensor/math.py
Comment (ZiadAmerr) [inline ivy/functional/frontends/paddle/tensor/math.py:74]:
```suggestion
```

**Review by ZiadAmerr — APPROVED** [review]

**Comment by rohinish404** [comment]
Thank you so much for the help @ZiadAmerr !!!

**Pull request merged**
